app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -215,21 +216,58 @@ class AddAdditionalMetadataController @Inject() (
 
 object AddAdditionalMetadataController {
   case class File(fileId: UUID, name: String)
+  val dateModifiedInsetText = "The date the record was last modified was determined during upload. This date should be checked against your own records: <strong>%s</strong>"
 
   def formFieldOverrides(formField: FormField, fileMetadata: Map[String, List[FileMetadata]]): FormField = {
-
+    formField.fieldId match {
+      case fieldId if fieldId == descriptionClosed => overrideDescriptionClosed(formField, fileMetadata)
+      case fieldId if fieldId == end_date          => overrideEndDate(formField, fileMetadata)
+      case fieldId if fieldId == closureStartDate  => overrideClosureStartDate(formField, fileMetadata)
+      case _                                       => formField
+    }
+  }
+  private def overrideDescriptionClosed(formField: FormField, fileMetadata: Map[String, List[FileMetadata]]): FormField = {
     // We have hard code this logic here as we are still not sure how to make it data-driven.
-    if (formField.fieldId == descriptionClosed) {
-      // Hide DescriptionClosed field if the Description property value is empty
-      val value = fileMetadata.get(description).map(_.head.value).getOrElse("")
-      val (fieldDescription, hideInputs, info) = if (value.isEmpty) {
-        ("If you need to add a description, you can do so in the Descriptive metadata step.", true, "")
-      } else {
-        ("The current description of your record is below. You can edit it in the Descriptive metadata step.", false, value)
-      }
-      formField.asInstanceOf[RadioButtonGroupField].copy(fieldDescription = fieldDescription, hideInputs = hideInputs, additionalInfo = info)
+    // Hide DescriptionClosed field if the Description property value is empty
+    val value = fileMetadata.get(description).map(_.head.value).getOrElse("")
```

Comment:
Maybe rename `value` to something more descriptive, for example: `val description`